support/cas-server-support-oidc/src/main/java/org/apereo/cas/oidc/config/OidcConfiguration.java

```diff
@@ -463,11 +467,23 @@ public OidcIntrospectionEndpointController oidcIntrospectionEndpointController()
         return new OidcIntrospectionEndpointController(context);
     }
 
+    /**
+     * @return
+     *  default implementation for {@link OidcPostLogoutRedirectUrlMatcher}: <pre>String::equalsIgnoreCase</pre>
+     */
+    @RefreshScope
+    @Bean(name = POST_LOGOUT_REDIRECTURL_MATCHER_BEAN_NAME)
+    @ConditionalOnMissingBean(name = POST_LOGOUT_REDIRECTURL_MATCHER_BEAN_NAME)
+    public OidcPostLogoutRedirectUrlMatcher postLogoutRedirectUrlMatcher() {
+        return String::equalsIgnoreCase;
+    }
+
     @RefreshScope
     @Bean
-    public OidcLogoutEndpointController oidcLogoutEndpointController() {
+    public OidcLogoutEndpointController oidcLogoutEndpointController(@Qualifier(POST_LOGOUT_REDIRECTURL_MATCHER_BEAN_NAME)
+                                                                         final OidcPostLogoutRedirectUrlMatcher postLogoutRedirectUrlMatcher) {
```

Comment:
Please invoke the bean directly inside the bean; not as an auto-wired parameter.

Reply:
? If I dont pass it as a bean, i cannot "overwrite" the matcher with another bean in our context - if I just call `postLogoutRedirectUrlMatcher()` inside of this its a method call which just cannot be replaced by another bean definition

Reply:
It should certainly be replacble. Proxies are used to wrap bean calls and you'd have a conditional bean that helps you supplant the default invocation. That's how everything else works in CAS as well. If you're seeing the opposite, please let me know and I'll take a look.

Reply:
you want the code be like 
```
    @RefreshScope
    @Bean
    public OidcLogoutEndpointController oidcLogoutEndpointController(@Qualifier(POST_LOGOUT_REDIRECTURL_MATCHER_BEAN_NAME)
                                                                         final OidcPostLogoutRedirectUrlMatcher postLogoutRedirectUrlMatcher) {
        val context = oidcConfigurationContext();
        return new OidcLogoutEndpointController(context, postLogoutRedirectUrlMatcher());
    }
```

right? Then its just a simple java method call to `postLogoutRedirectUrlMatcher()` in the same class and has nothing do do with Spring dependency injection, one might as well remove the bean annotation from the method called (like I said, this is already a problem with the OidcConfigurationContext.

otherwise I dont know what you mean

Reply:
Yes that how it should be. `postLogoutRedirectUrlMatcher()` should be wrapped in a proxy that would intercept it on subsequent calls. It can and should be overridden in custom configuration contexts, since the bean is marked as conditional. The proxy, in a conditional context should call the custom bean definition if it's defined, and otherwise, would route the call to the original bean/method definition.

Reply:
wow I didnt know that but I tested it and it really works like you said - Spring is even more amazing/black-magic (still undecided) then i thought, thanks for the explanation!!

Reply:
You're quite welcome :) Thanks for the confirmation.